keps/sig-auth/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1003 @@
+-->
+
+Add new subresources for `healthz` and `pods` endpoint while supporting the old coarse grained
+proxy subresources as shown in the table below.
+
+| Request path                                     | Existing Subresource | Proposed Subresource(s) |
+| ----------------------------------------------- | -------------------- | ------------------------ |
+| /healthz                                        | proxy                | proxy, healthz           |
+| /healthz/log                                    | proxy                | proxy, healthz           |
+| /healthz/ping                                   | proxy                | proxy, healthz           |
+| /healthz/syncloop                               | proxy                | proxy, healthz           |
+| /pods/                                          | proxy                | proxy, pods              |
+| /runningpods/                                   | proxy                | proxy, pods              |
+
+This way users who were previously using the `proxy` verb to access `/healthz`
+kubelet endpoint can update their `ClusterRole` or `Role` by replacing the
+`proxy` verb with the `healthz` verb. 
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a security conscious node monitoring agent owner I want to interact with the
+kubelet API to list pods on the node without having to grant the agent the `proxy` subresource and use
+least privilege to grant the exact permissions the agent needs like `pods` or 
+`healthz`.
+
+### Notes
+
+If this change were to be implemented as proposed, the `proxy` `subresource` 
+would still cover `/attach/`, `/exec/`, `/run/`, `/configz/`, `/debug/`. 
+The reader might be wondering why we didn't break these permissions out into 
+their own `subresource`?
+
+The reason why we did not breakout all endpoints under `proxy` into their own
+`subresource` is because we could not reason about a case where if we allowed
+one of these permissions, having any of the other permissions would be considered
+worse. If you have `/exec/` then having `/run/` or `/attach/` isn't making it 
+worse for an attacker.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Since we are adding a second `SubjectAccessReview` request the latency for some
+requests will increase. However, `kubelet` has a `SubjectAccessReview` request
+cache, so subsequent requests should result in cache hits.
```

Comment:
```suggestion
Since we are adding a second SubjectAccessReview request the latency for some
requests will increase. However, `kubelet` has a SubjectAccessReview` response
cache, so subsequent requests should result in cache hits.
```